frontend/src/components/common/SearchAndFilters.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect, useState } from "react";
+import {
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionPanel,
+  AccordionIcon,
+  Box,
+  Grid,
+  GridItem,
+  Button,
+  Checkbox,
+  FormControl,
+  FormLabel,
+  Input,
+  InputLeftElement,
+  Icon,
+  InputGroup,
+  Select,
+  Divider,
+  Text,
+} from "@chakra-ui/react";
+import { Card } from "react-bootstrap";
+import commonAPIClient from "../../APIClients/CommonAPIClient";
+import { LogRecord } from "./types/LogRecord";
+
+type Props = {
+  setLogRecords: React.Dispatch<React.SetStateAction<LogRecord[]>>;
+};
+
+const SearchAndFilters = ({ setLogRecords }: Props): React.ReactElement => {
+  /* TODO: change inputs to correct types
+  - this is possible when the data in LogRecords comes from the API since the API will have the employees/attnTos names + userId
+    names + userId and we can query using the /users route based on that
+    e.g.:
+      - for employee -> we'll need to get an employee's ID by their name to pass into the route
+      - for attention to -> same as above
+      - for startDate & endDate the date format should be 'YYYY-MM-DD'
+       - for tags -> change to an actual array of strings instead of a string
+  */
+  // TODO: search by resident
+  const [residents, setResidents] = useState("");
+  const [employees, setEmployees] = useState("");
+  const [startDate, setStartDate] = useState("");
+  const [endDate, setEndDate] = useState("");
+  const [tags, setTags] = useState("");
+  const [attentionTo, setAttentionTo] = useState("");
+  const [building, setBuilding] = useState("");
+  const [flagged, setFlagged] = useState(false);
+
+  useEffect(() => {
+    const getLogRecordsAfterFiltering = async () => {
+      const employeeIds = employees
+        ? employees.replaceAll(`"`, "").split(",")
+        : [];
+      const attentionTos = attentionTo
+        ? attentionTo.replaceAll(`"`, "").split(",")
+        : [];
+      const dateRange = startDate && endDate ? [startDate, endDate] : [];
+
+      const data = await commonAPIClient.filterLogRecords(
+        building,
+        employeeIds,
+        attentionTos,
+        dateRange,
+        tags ? [tags] : [],
+        flagged,
+      );
+      setLogRecords(data);
+    };
+
+    getLogRecordsAfterFiltering();
+  }, [
+    building,
+    employees,
+    attentionTo,
+    startDate,
+    endDate,
+    tags,
+    flagged,
+    setLogRecords,
+  ]);
+
+  // console.log("data", data);
+  return (
+    <Card>
+      <Box padding="8px 16px 20px">
+        <Text fontSize="12px" fontWeight="700" color="#6D8788" align="left">
+          FILTER BY
+        </Text>
+        <FormControl>
+          <Grid padding="10px 0px" templateColumns="repeat(7, 1fr)" gap={7}>
+            <GridItem colSpan={2}>
```

Comment:
Thought: Should we specify these entries should be formatted like this here. Like "eg. ______" 

<img width="593" alt="Screen Shot 2023-05-22 at 1 07 50 AM" src="https://github.com/uwblueprint/supportive-housing/assets/64047863/92d771d5-5bec-48a0-aacc-7cd3ddef6f7b">

Reply:
as in change the placeholders to the designs? right now we can't filter by employee name or resident actual ID # so i think it could be confusing for devs. in the future you'd have to link names -> ids